server/server.go

```diff
@@ -415,20 +431,62 @@ func (s *Server) v0QueryPath(w http.ResponseWriter, r *http.Request, path ast.Re
 	writer.JSON(w, 200, qrs[0].Result, false)
 }
 
-func (s *Server) v1DataGet(w http.ResponseWriter, r *http.Request) {
+func (s *Server) returnDiagnostics(w http.ResponseWriter, r *http.Request) {
+	pretty := getBoolParam(r.URL, types.ParamPrettyV1, true)
+	explainMode := getExplain(r.URL.Query()[types.ParamExplainV1], types.ExplainTruthV1)
 
+	s.buf.Lock()
+	defer s.buf.Unlock()
+
+	var results []types.DiagnosticsResponseElementV1
+	s.diagnostics.Iter(func(info *diagnostics.Info) bool {
+		result := types.DiagnosticsResponseElementV1{ID: info.ID, Query: info.Query}
+
+		if info.Metrics != nil {
+			result.Metrics = info.Metrics.All()
+		}
+
+		if info.Trace != nil {
+			result.Explanation = s.getExplainResponse(explainMode, info.Trace)
+		}
+
+		switch r := info.Results.(type) {
+		case error:
+			result.Error = types.NewErrorV1(types.CodeInternal, r.Error())
+		default:
+			result.Result = r
+		}
+
+		results = append(results, result)
+		return false
+	})
+
+	writer.JSON(w, 200, results, pretty)
+}
+
+func (s *Server) v1DataGet(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	vars := mux.Vars(r)
 	path := stringPathToDataRef(vars["path"])
+	settings := s.evalDiagnosticPolicy(r)
 
+	sendDiagnostics := path.String() == "data.system.diagnostics"
```

Comment:
Did you consider adding a new route to the server for /v1/data/system/diagnostics? This way I'd expect the v1DataGet handler doesn't have to be overloaded like this.